streamer/src/recvmmsg.rs

```diff
@@ -126,11 +142,24 @@ pub fn recv_mmsg(sock: &UdpSocket, packets: &mut [Packet]) -> io::Result</*num p
         usize::try_from(nrecv).unwrap()
     };
     for (addr, hdr, pkt) in izip!(addrs, hdrs, packets.iter_mut()).take(nrecv) {
-        pkt.meta_mut().size = hdr.msg_len as usize;
-        if let Some(addr) = cast_socket_addr(&addr, &hdr) {
+        let hdr_ref = unsafe { hdr.assume_init_ref() };
+        let addr_ref = unsafe { addr.assume_init_ref() };
+        pkt.meta_mut().size = hdr_ref.msg_len as usize;
+        if let Some(addr) = cast_socket_addr(&addr_ref, hdr_ref) {
             pkt.meta_mut().set_socket_addr(&addr);
         }
     }
+
+    // TODO - figure out in review whether we think we have to drop these OR if
+    //        leaving as MaybeUninit (no drop) is ok
+    for (iov, addr, hdr) in izip!(&mut iovs, &mut addrs, &mut hdrs).take(count) {
+        unsafe {
+            iov.assume_init_drop();
+            addr.assume_init_drop();
+            hdr.assume_init_drop();
+        }
+    }
```

Comment:
I figured it'd be easier to discuss this here by including it on the diff.
- Unlike the previous PR, the arrays have the potential to be a mix of initialized / uninitialized. This could happen if `packets.len() != NUM_RCVMMSGS`. So, we can't transmute the entire array from `MaybeUninit<T>` to `T` like the other PR did
- That being said, I'm not convinced we need to drop these items. The structs have primitive integers and raw pointers; the raw pointers are to elements on this stack frame OR to `packets` (which has a separate lifetime anyways)
    
So, I'm inclined to remove this block. But, commenting here for second set of eyes to make sure I'm not overlooking anything

Reply:
As you said, "memory wise" this won't leak if we don't drop because it's a bunch
of aggregates of primitives that do nothing on drop. However this is C thinking
:P

In the rust memory model, if something is initialized it _must_ be dropped or
conceptually it's a leak, and I'm fairly sure that sanitizers would catch this. 

I would drop, changing count to count.min(nrecv)

Reply:
Fair enough - I'll drop and add a safety note; let's discuss `count` vs. `nrecv` in the other issue